src/common/lockdep.cc

```diff
@@ -41,9 +41,8 @@ static constexpr size_t MAX_LOCKS = 128 * 1024;   // increase me as needed
 static std::bitset<MAX_LOCKS> free_ids; // bit set = free
 static ceph::unordered_map<pthread_t, std::map<int,ceph::BackTrace*> > held;
 static constexpr size_t NR_LOCKS = 4096; // the initial number of locks
-static constexpr size_t MAX_FOLLOWERS = 4096;
-static std::vector<std::bitset<MAX_FOLLOWERS>> follows(NR_LOCKS); // follows[a][b] means b taken after a
-static std::vector<std::array<ceph::BackTrace *, MAX_FOLLOWERS>> follows_bt(NR_LOCKS);
+static std::vector<std::bitset<MAX_LOCKS>> follows(NR_LOCKS); // follows[a][b] means b taken after a
```

Comment:
i don't understand why s/MAX_FOLLOWERS/MAX_LOCKS/. could you expand a little bit on it?

Reply:
ahh, i see. the followers can still be distributed to the space higher than 4096, even the total number of which should be much smaller than 4096.

Reply:
`follows[p->first].set(id); `

The maximum of the `id` could reach to `MAX_LOCKS - 1`.